src/common/queries/documents.ts

```diff
@@ -28,3 +29,16 @@ export function useDocumentsQuery(params: Params) {
     )
   );
 }
+
+export const useDocumentsBulk = () => {
+  return (ids: string[], action: 'download', onActionCall?: () => void) => {
+    toast.processing();
+
+    request('POST', endpoint('/api/v1/documents?per_page=100'), {
+      action,
+      ids,
+    })
+      .then(() => toast.success('exported_data'))
+      .finally(() => onActionCall?.());
+  };
+};
```

Comment:
I think this is an antipattern. You should be returning empty `Promise` at least letting action know it's successful rather than taking your own scope of callback fn.

Reply:
@beganovich Until now, we've had a situation where even after the successful completion of a specific bulk action, the selected entities remained checked. What I wanted to achieve with this is to improve the UX during the execution of bulk actions. So, when a certain action is performed, we will perform a deselection of all selected entities. In this case, the function `onActionCall` is essentially a reference to this function:

```
 const onBulkActionCall = () => {
    if (mainCheckbox.current) {
      mainCheckbox.current.checked = false;

      setSelected([]);
    }
  };
```

Actually, it's the same thing you did for `archive/restore/delete` bulk actions in the DataTable component under the `finally` block. However, the issue is that the execution of custom bulk actions is not available under this component:

```
.finally(() => {
        queryClient.invalidateQueries([props.endpoint]);
        queryClient.invalidateQueries([apiEndpoint.pathname]);

        setSelected([]);
      });
```

The `onBulkActionCall` is passed through the `bulkAction` function and is available for use in `customBulk` hooks to perform bulk actions.
  
So, once the action is done, deselecting elements will make it much easier to continue using bulk actions. That's sort of my idea.

Now I'm getting the idea that maybe this could be achieved through a `custom event`, or through `atoms`, or via `Redux`.

What are your thoughts on those additional implementation methods?